File: GO/Microondas.go
package main

import "fmt"
type Microondas struct {
	Electrodomestico
	Potencia int
	Sirve    bool
}
//Se crea el metodo constructor de microondas
func NewMicroondas(potencia int, sirve bool) *Microondas {
	return &Microondas{
		Potencia: potencia,
		Sirve:    sirve,
	}
}
// Implementamos los métodos abstractos
func (m *Microondas) Encender(encendido bool) {
	m.encendido = encendido
	if encendido {
		fmt.Println("encendido")
	} else {
		fmt.Println("Microondas apagado")
	}
}

func (m Microondas) Funcionar() {
	if m.encendido {
		fmt.Println("El microondas está calentando la comida")
	} else {
		fmt.Println("El microondas no funciona")
	}
}

File: C#/Microondas.cs
using System;

namespace poocorredor
{
    public class Microondas : Electrodomestico
    {
        private bool sirve;
        private string marca;

        public Microondas(bool sirve, string marca)
        {
            this.sirve = sirve;
            this.marca = marca;
        }
        //Metodos heredados de la clase electrodomestico que son abstractos
        public override void encender(bool encendido)
        {
            if (encendido == true)
            {
                Console.WriteLine("Se enciende");
            }
            else
            {
                Console.WriteLine("se apaga");
            }
        }

        public override void funcionar()
        {
            Console.WriteLine("Funciono");
        }
    }
}
